2/Dockerfile

```diff
@@ -1,21 +1,41 @@
 FROM buildpack-deps:jessie
 
-# remove several traces of debian python
-RUN apt-get purge -y python.*
+# ensure local pypy is preferred over distribution python
+ENV PATH /usr/local/bin:$PATH
 
 # http://bugs.python.org/issue19846
 # > At the moment, setting "LANG=C" on a Linux system *fundamentally breaks Python 3*, and that's not OK.
 ENV LANG C.UTF-8
 
-ENV PYPY_VERSION 5.4.0
+# runtime dependencies
+RUN apt-get update && apt-get install -y --no-install-recommends \
+		tcl \
+		tk \
+	&& rm -rf /var/lib/apt/lists/*
 
-RUN set -x \
-	&& curl -SL "https://bitbucket.org/pypy/pypy/downloads/pypy2-v${PYPY_VERSION}-linux64.tar.bz2" \
-		| tar -xjC /usr/local --strip-components=1
+ENV PYPY_VERSION 5.4.0
+ENV PYPY_SHA256SUM bdfea513d59dcd580970cb6f79f3a250d00191fd46b68133d5327e924ca845f8
```

Comment:
It's from the downloads page: http://pypy.org/download.html#checksums

I didn't realise this stuff was updated via a script. I could work something into `update.sh` that does something like this (accounting for versions and such):
```bash
curl http://pypy.org/download.html \
  | grep -E '[a-f0-9]{64}  pypy2-v5.4.0-linux64.tar.bz2' \
  | cut -d' ' -f1
```

What do you think? Or is it just not worth the effort?